src/amo/components/AddonReview/index.js

```diff
@@ -34,25 +37,34 @@ type RefreshAddonFunction = (
 type UpdateReviewTextFunction =
   (review: $Shape<SubmitReviewParams>) => Promise<void>;
 
+type DispatchMappedProps = {|
+  refreshAddon?: RefreshAddonFunction,
+  setDenormalizedReview?: SetDenormalizedReviewFunction,
+  updateReviewText?: UpdateReviewTextFunction,
+|};
+
 type Props = {|
-  apiState: ApiStateType,
-  createLocalState: typeof defaultLocalStateCreator,
-  debounce: typeof defaultDebounce,
-  errorHandler: ErrorHandlerType,
-  i18n: I18nType,
+  createLocalState?: typeof defaultLocalStateCreator,
+  debounce?: typeof defaultDebounce,
   onEscapeOverlay?: () => void,
   onReviewSubmitted: () => void | Promise<void>,
-  refreshAddon: RefreshAddonFunction,
   review: UserReviewType,
-  setDenormalizedReview: SetDenormalizedReviewFunction,
-  updateReviewText: UpdateReviewTextFunction,
+  ...DispatchMappedProps,
 |};
 
+type InjectedProps = {|
+  apiState: ApiStateType,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+|};
+
+type InternalProps = { ...Props, ...InjectedProps };
+
 type State = {|
   reviewBody: ?string,
 |};
 
-export class AddonReviewBase extends React.Component<Props, State> {
+export class AddonReviewBase extends React.Component<InternalProps, State> {
```

Comment:
Why is this type `Internal`? I am not sure to understand. I see that we often use `CompBase` as class name, so maybe it could be named `BaseProps`. It does not really matter though.

Reply:
They are internal because only the component will see things like the mapped props. No calling code will need to pass in these props.